### DISCUSSION TITLE
Cursor tracking in Editor

### DISCUSSION THREAD

**Discussion by leo-paz** [GrapesJS/grapesjs#3736] [Q&A]
Hi @artf and GrapesJS Community,

I'm trying to add ruler functionality and have used the code here as a resource: https://github.com/Ju99ernaut/grapesjs-rulers#readme

The code there works but I'm trying to extend the ruler functionality so you actually see lines follow your cursor when you active the ruler command like so: 
![image](https://user-images.githubusercontent.com/26093865/131574713-0b7e7941-3641-413a-a743-0c1ef1e0332b.png)

It works while the cursor is outside of the body of the canvas but once I hover over the canvas its rendered positions has an offset and that offset gets smaller as I glide to the left, Shown in the GIF below:
![mouse-capture](https://user-images.githubusercontent.com/26093865/131576373-82a29176-d6c5-4507-bbaf-a03b00265802.gif)

One thing that caused the ruler lines to actually snap to the right spot was when the cursor detected a pointer event if that helps. Would love some direction as to what solutions to explore next ❤️

**Comment by artf** [comment]
I'd suggest asking for help from the maintainer of the plugin.
